app/downloader/inspector.py

```diff
@@ -64,3 +79,52 @@ def get_latest_date(self):
             return None
         else:
             return max(dates)
+
+
+class SeleniumOnlineInspector(OnlineInspector):
+    def __init__(self, url, download_link_regex, download_link_xpath, file_date_format='%Y-%m-%d'):
+        self.download_link_xpath = download_link_xpath
+        super().__init__(url, download_link_regex, file_date_format)
+
+    def get_all_data_urls(self):
+        chrome_options = webdriver.ChromeOptions()
+        chrome_options.add_argument('--headless')
+        chrome_options.add_argument('--no-sandbox')
+        chrome_options.binary_location = flask_app.config['app']['chrome_binary_location']
+        ignored_exceptions = (
+            NoSuchElementException,
+            StaleElementReferenceException,
+        )
+        driver = webdriver.Chrome(
+            executable_path=flask_app.config['app']['chrome_driver_path'],
+            chrome_options=chrome_options,
+        )
+        driver.get(self.url)
+
+        class page_has_element:
+            def __init__(self, xpath):
+                self.xpath = xpath
+
+            def __call__(self, driver):
+                elements = driver.find_elements_by_xpath(self.xpath)
+                if len(elements) > 0:
+                    return elements
+                return False
+
+        found = False
+        try:
+            elements = WebDriverWait(driver, 10, ignored_exceptions=ignored_exceptions).until(
+                page_has_element(self.download_link_xpath)
+            )
```

Comment:
Could this be simplified to just the following? It doesn't look like you need a class for it - but maybe there's something I'm missing?

```suggestion
        found = False
        try:
            elements = WebDriverWait(driver, 10, ignored_exceptions=ignored_exceptions).until(
                lambda driver: driver.find_elements_by_xpath(self.download_link_xpath)
            )
```

Reply:
yeah, you're right. i first had something more complex in there but now it can be replaced with just a lambda expression. good spot.